File: atcoder.jp/PyPy3 (7.3.0)/abc245/abc245_c.py
N, K = map(int,input().split())
A = list(map(int,input().split()))
B = list(map(int,input().split()))

dp = [False] * (N)
ep = [False] * (N)
dp[0] = ep[0] = True

for i in range(1,N):
    if dp[i-1]:
        if abs(A[i-1] - A[i]) <= K:
            dp[i] = True
        if abs(A[i-1] - B[i]) <= K:
            ep[i] = True
    if ep[i-1]:
        if abs(B[i-1] - A[i]) <= K:
            dp[i] = True
        if abs(B[i-1] - B[i]) <= K:
            ep[i] = True

if dp[-1] or ep[-1]:
    print('Yes')
else:
    print('No')

File: atcoder.jp/Nim (1.0.6)/abc245/abc245_c.nim
import strutils
import sequtils
import strformat
import future

var N, K: int
(N, K) = stdin.readLine.split.map(parseInt)
var 
    A = stdin.readLine.split.map(parseInt)
    B = stdin.readLine.split.map(parseInt)
    dp:seq[bool] = repeat(false, N)
    ep:seq[bool] = repeat(false, N)

dp[0] = true
ep[0] = true

for i in 1..N-1:
    if dp[i-1]:
        if abs(A[i-1] - A[i]) <= K:
            dp[i] = true
        if abs(A[i-1] - B[i]) <= K:
            ep[i] = true
    if ep[i-1]:
        if abs(B[i-1] - A[i]) <= K:
            dp[i] = true
        if abs(B[i-1] - B[i]) <= K:
            ep[i] = true

if dp[N-1] or ep[N-1]:
    echo "Yes"
else:
    echo "No"